addon/utils/locale.js

```diff
@@ -87,10 +87,11 @@ export default class Locale {
   }
 
   _defineMissingTranslationTemplate(key) {
+    const I18n = this.container.lookupFactory('service:i18n');
     const missingMessage = this.container.lookupFactory('util:i18n/missing-message');
     const locale = this.id;
 
-    function missingTranslation(data) { return missingMessage(locale, key, data); }
+    function missingTranslation(data) { return missingMessage.call(I18n, locale, key, data); }
```

Comment:
Since we're going to call `missingMessage` on the` service:i18n`, why don't we just add it to that object's public API:
```js
// services/i18n:
{
  buildMissingTranslation(locale, key, data) {
    const fn = this.container.lookupFactory('util:i18n/missing-message');
    return fn.call(this, locale, key, data);
  }
}

// locale.js:
{
  _defineMissingTranslationTemplate(key) {
    ...
    function missingTranslation(data) {
      return i18n.buildMissingTranslation(locale, key, data);
    }
    ...
  }
}
```